### PR TITLE
Renaming BaseControllerView to BaseView.

### PR DESCRIPTION
**Pull request by tomasperezv**
### What is the goal?

We need to rename `BaseControllerView` to `BaseView`

- [x] The `README.md` has been updated accordingly

### Reviewers

@jobandtalent/frontend

### CODE REVIEW

File: src/javascript/component/base/base-controller.js [hunk 1]
```diff
@@ -133,7 +133,7 @@ class BaseController {
       return result;
     });
 
-    return viewName.join('');
+    return viewName.join('').replace('controller-', '');
```
Comment (montiu) [inline src/javascript/component/base/base-controller.js]: same here

File: src/javascript/component/base/base-controller.js [hunk 2]
```diff
@@ -75,13 +75,13 @@ class BaseController {
...
    */
-  _requireAndGetViewClass(): Class<BaseControllerView> { // eslint-disable-line
+  _requireAndGetViewClass(): Class<BaseView> { // eslint-disable-line
     let dependency = null;
 
     /* global modulejs */
     // $FlowFixMe: modulejs is not defined
     if (typeof modulejs !== 'undefined') {
-      dependency = modulejs.require(`${this._id}View`);
+      dependency = modulejs.require(`${this._id.replace('Controller', '')}View`);
```
Comment (montiu) [inline src/javascript/component/base/base-controller.js]: we keep using "Controller" naming inside ?

Comment (tomasperezv) [inline src/javascript/component/base/base-controller.js]: @montiu notice that this is the `BaseController`, not the `BaseView`. The `BaseController` loads dynamically its associated view by using the `_id` property that it has. That property contains the `Controller` string that we need to remove in two places, that's what you are seeing here but there is any reference left whatsoever to the deprecated `BaseControllerView`.

For that reason we need to remove `Controller` w

Comment (montiu) [inline src/javascript/component/base/base-controller.js]: 👍

**Comment by calderon** [comment]
O🔑

File: src/component/overlay/overlay-view.coffee [hunk 3]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-BaseView = require('../../javascript/component/base/base-controller-view').default
+BaseView = require('../../javascript/component/base/base-controller').default
```
Comment (alejandroweb) [inline src/component/overlay/overlay-view.coffee]: Typo here remove `controller` not `view`

File: src/component/popup/popup-view.coffee [hunk 4]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-BaseView = require('../../javascript/component/base/base-controller-view').default
+BaseView = require('../../javascript/component/base/base-controller').default
```
Comment (alejandroweb) [inline src/component/popup/popup-view.coffee]: Typo here remove `controller` not `view`

File: src/core.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ window.jQuery = window.$ = require('./javascript/dom/core');
 // ModuleJS modules
 const ModuleJS = require('@lrsjng/modulejs');
 const BaseController = require('./javascript/component/base/base-controller').default;
-const BaseView = require('./javascript/component/base/base-controller-view').default;
+const BaseView = require('./javascript/component/base/base-controller').default;
```
Comment (alejandroweb) [inline src/core.js]: Typo here remove `controller` not `view`

**Comment by alejandroweb** [comment]
👍

**Pull request merged**